lib/ddtrace/contrib/faraday/middleware.rb

```diff
@@ -0,0 +1,52 @@
+require 'faraday'
+require 'ddtrace/ext/http'
+require 'ddtrace/ext/net'
+
+module Datadog
+  module Contrib
+    module Faraday
+      # Middleware implements a faraday-middleware for ddtrace instrumentation
+      class Middleware < ::Faraday::Middleware
+        def initialize(*)
+          @tracer = try_pin(:tracer, Datadog.tracer)
+          @service = try_pin(:service, SERVICE)
+
+          super
+        end
+
+        def call(env)
+          tracer.trace(SERVICE) do |span|
```

Comment:
This should be the span name and not the service. Something like `faraday.request` and not related to what is the default service name